niljs/src/utils/transaction.ts

```diff
@@ -0,0 +1,70 @@
+import type { PublicClient } from "../clients/PublicClient.js";
+import { bytesToHex } from "../encoding/fromBytes.js";
+import type { Hex } from "../types/Hex.js";
+import type { ProcessedReceipt } from "../types/IReceipt.js";
+import { waitTillCompleted } from "./receipt.js";
+
+/**
+ * Represents a transaction that can be awaited for completion
+ */
+export class TransactionResponse {
+  public readonly hash: Hex;
+  constructor(
+    private readonly tx: Uint8Array,
+    private readonly client: PublicClient,
+    private readonly shardId: number,
```

Comment:
Yo don't need shardId here. Shard id also part of tx. Also it should be still be awaitable by waitTillCompleted (and you should update this method also to work with Hex and Transaction) and it should be upgraded.